app/src/main/java/com/example/househomey/HomeFragment.java

```diff
@@ -236,4 +279,56 @@ private void updateListData() {
         this.listSumView.setText("$" + listSum.toString());
         this.listCountView.setText(Integer.toString(listCount));
     }
+
+    /**
+     * Displays the sort menu with properties of items to sort by
+     * @param view The view on which the pop up menu is displayed
+     * @param sortProperties A mapping of properties to their comparators
+     */
+    private void showSortMenu(View view, Map<String, Comparator<Item>> sortProperties) {
+        PopupMenu popupMenu = new PopupMenu(requireContext(), view);
+        MenuInflater inflater = popupMenu.getMenuInflater();
+        inflater.inflate(R.menu.sort, popupMenu.getMenu());
+
+        sortOrder = toggleOrder.isChecked();
+        popupMenu.setOnMenuItemClickListener(item -> {
+            int itemId = item.getItemId();
+            if (itemId == R.id.sort_by_description) {
+                currentSort = sortProperties.get("description");
+                sortItems();
+            } else if (itemId == R.id.sort_by_date) {
+                currentSort = sortProperties.get("date");
+                sortItems();
+            } else if (itemId == R.id.sort_by_make) {
+                currentSort = sortProperties.get("make");
+                sortItems();
+            } else if (itemId == R.id.sort_by_estimatedValue) {
+                currentSort = sortProperties.get("cost");
+                sortItems();
```

Comment:
Another nitpick (sorry), we could move all these sortItems() to just one call at the bottom.